plugins/inputs/execd/shim/goshim_windows.go

```diff
@@ -13,6 +13,7 @@ func listenForCollectMetricsSignals(ctx context.Context, collectMetricsPrompt ch
 	signal.Notify(collectMetricsPrompt, syscall.SIGHUP)
 
 	go func() {
+		//nolint:gosimple // for-select used on purpose
```

Comment:
Hmmm there is no `for` so I wonder if the following code for the go-routine would also work
```golang
	go func() {
		<-ctx.Done()

		// context done. stop to signals to avoid pushing messages to a closed channel
		signal.Stop(collectMetricsPrompt)
	}()
```